### DISCUSSION TITLE
Any plans for the watch api?

### DISCUSSION THREAD

**Discussion by robertavram** [ory/keto#551] [Ideas]
From the Zanzibar paper:

_2.4.3 Watch
Some clients maintain secondary indices of relation tuples
in Zanzibar. They can do so with our Watch API. A watch
request specifies one or more namespaces and a zookie representing the time to start watching. A watch response contains all tuple modification events in ascending timestamp
order, from the requested start timestamp to a timestamp encoded in a heartbeat zookie included in the watch response.
The client can use the heartbeat zookie to resume watching
where the previous watch response left off._

**Comment by zepatrik** [comment]
There are no concrete plans yet. If you have any use cases or ideas, feel free to collect them here.

**Comment by robertavram** [comment]
In our case, we have quite a few systems, most, if not all of them seem to manage the access control in-app (no external api calls, locally managed - you can imagine the nightmare). 
There is low appetite to change the way access control works for some of these apps, but high appetite for managing acls centrally. 

We need a way (we're probably looking at creating a few libs) to make sure that we update the local access control mechanisms based on a central source of truth and some fancy mapping.

We recognize that it's not the best way to go about this but we need something workable in place.
